astropy/units/tests/test_equivalencies.py

```diff
@@ -791,6 +791,32 @@ def test_pixel_scale():
     assert_quantity_allclose(asec.to(u.pix, u.pixel_scale(pixscale2)), pix)
 
 
+def test_pixel_scale_invalid_scale_unit():
+
+    pixscale = 0.4 * u.arcsec
+
+    with pytest.raises(Exception):
```

Comment:
Maybe be a bit more precise in catching the exception: `pytest.raises(u.UnitsError, match="pixel dimension")`

I'd also add one with `u.pix**2` or so.

Reply:
Good suggestion. Added a new case with unexpected u.pix dimensionality.